### PR TITLE
Adjust T1 frontend to claim store change

### PR DESCRIPTION
**Pull request by rlewan**
### JIRA link ###

https://tools.hmcts.net/jira/browse/ROC-2179

### Change description ###

Updates for T1 frontend to work with unified claim store model.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/claims/claimModelConverter.ts [hunk 1]
```diff
@@ -14,12 +14,15 @@ export class ClaimModelConverter {
...
 
-    draftClaim.claimant.dateOfBirth = draftClaim.claimant.dateOfBirth.date.asString() as any
     draftClaim.reason = draftClaim.reason.reason as any
 
+    draftClaim['feeAmountInPennies'] = draftClaim.claimant.payment.amount
+
     draftClaim['payment'] = draftClaim.claimant.payment as any
     delete draftClaim.claimant.payment
 
+    draftClaim.amount['type'] = 'breakdown'
```
Comment (yarpo) [inline src/main/app/claims/claimModelConverter.ts]: don't we have any enum for that?

File: src/main/app/claims/claimModelConverter.ts [hunk 2]
```diff
@@ -36,6 +39,8 @@ export class ClaimModelConverter {
     draftClaim.claimant['type'] = 'individual'
 
     draftClaim.claimant.mobilePhone = draftClaim.claimant.mobilePhone.number as any
+
+    draftClaim.claimant.dateOfBirth = draftClaim.claimant.dateOfBirth.date.asString() as any
```
Comment (yarpo) [inline src/main/app/claims/claimModelConverter.ts]: don't we always want to keep dates as Moment?

File: src/main/app/claims/claimModelConverter.ts [hunk 3]
```diff
@@ -58,6 +63,9 @@ export class ClaimModelConverter {
     }
 
     draftClaim.defendant['type'] = 'individual'
```
Comment (yarpo) [inline src/main/app/claims/claimModelConverter.ts]: enum?

File: src/main/app/claims/claimModelConverter.ts (hunk 1 above)
Comment (timja-kainos) [reply src/main/app/claims/claimModelConverter.ts]: nope its not an enum, easier to just hardcode it here I would say, we shouldn't ever have to look at it.

File: src/main/app/claims/claimModelConverter.ts (hunk 2 above)
Comment (timja-kainos) [reply src/main/app/claims/claimModelConverter.ts]: this function converts to moment and then formats it how as want it, asString() may not be the best name, but it is what we want to do here
```
this.toMoment().format(DATE_FORMAT)
```

File: src/main/app/claims/claimModelConverter.ts (hunk 3 above)
Comment (timja-kainos) [reply src/main/app/claims/claimModelConverter.ts]: anirudha's PR has this completely reworked, this is temporary

File: src/main/app/claims/models/claimData.ts [hunk 4]
```diff
@@ -8,18 +8,30 @@ import Payment from 'app/pay/payment'
...
+      throw new Error('This claim has multiple defendants')
+    }
+  }
+
   deserialize (input: any): ClaimData {
     if (input) {
       this.claimant = new Claimant().deserialize(input.claimant)
-      this.defendant = new Defendant().deserialize(input.defendant)
+      let defendants: Defendant[] = []
+      input.defendants.forEach((defendant: Defendant) => {
```
Comment (timja-kainos) [inline src/main/app/claims/models/claimData.ts]: this should be a map

**Pull request merged**